zyte_common_items/util.py

```diff
@@ -1,7 +1,13 @@
-from typing import Any, Callable, Dict, Optional, Tuple, Type
+from typing import Any, Callable, Dict, Optional, Tuple, Type, Union
 from weakref import WeakKeyDictionary
 
 import attr
+from web_poet.page_inputs.url import _Url
+
+try:
+    from typing import TypeGuard  # for Python 3.9+
```

Comment:
https://peps.python.org/pep-0647/ states it's Python 3.10 - is the feature available in 3.9 as well?

Reply:
Nice catch! Fixed in https://github.com/zytedata/zyte-common-items/pull/11/commits/391f617e502ba98e5abbf701c2e6243d8b15f9ad

Reply:
do you need to fix setup.py as well?

Reply:
Ahhh missed that one. I was under the impression I was using less than or equal. 😅 . Updated in https://github.com/zytedata/zyte-common-items/pull/11/commits/ec525edea85227932086c341630d277836b51d81

Reply:
for me it's not clear why were tests passing in Python 3.9

Reply:
For Python 3.9, importing `TypeGuard` from `typing` would error out and it'd be imported via `typing_extensions` instead.

I also realized that https://pypi.org/project/typing-extensions/ also works until 3.10.

Reply:
How would it be imported from type_extensions in Python 3.9, if they were only installed for Python < 3.9?

Reply:
I see 🤔 

Could it be due to GitHub Action's cache? In this commit https://github.com/zytedata/zyte-common-items/commit/3e53d6154ce7e5f1f9d0ec85f7768c100de3ad7f#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7R26, it was installed on all versions before `python_version<'3.9'` was incorrectly specified.

Reply:
I took a look into this again, trying it out locally while ensuring to delete all caches but `typing-extensions` was still being installed.

Turns out that `pytest-my-testing` is the one that installs it. 😅 https://github.com/davidfritzsche/pytest-mypy-testing/blob/031514ff6ecd5bdf4d11ff238c14d4801b5e47f3/requirements.txt#L169

Reply:
Thanks for checking it @BurnzZ!